testcase/integration/test_integration.py

```diff
@@ -0,0 +1,281 @@
+
+            print("REQ url: ", url)
+            print("RES block: ", block)
+
+            assert "error" not in block
+
+    # @pytest.mark.skip
+    def test_send_tx_message(self, request):
+        """Test for 'send_transaction'
+
+        Note:
+            Uses Message Type Transaction.
+            From address is equal to To address.
+            Interval await time is essential, due to consensus completion and different tx hashes between channels.
+
+        Test steps:
+            1. Get peer info from first peer
+            2. Extract key and password and make wallet
+            3. Build message transaction and sign it
+            4. Send Tx to first channel.
+            5. Await consensus time(currently 0.5 * <<Number of peers>> 'sec')
+            6. Repeat from '3'
+
+        Assertion Test:
+            1. Check that return value of send_transaction has valid tx hash format.
+        """
+        global peer_conf_path_list
+
+        channel_count = int(request.config.getoption("--channel-count"))
+
+        from_peer = conftest.get_peer_info(conf_path_list=peer_conf_path_list, order=0)
+        key_path = from_peer["PRIVATE_PATH"]
+        key_pass = from_peer["PRIVATE_PASSWORD"]
+        wallet = KeyWallet.load(key_path, key_pass)
+
+        for channel_order in range(channel_count):
+            # Create message
+            byte_msg = f"test_msg on {random.randint(0, 44444)}".encode("utf-8")
+            hex_msg = b"0x" + binascii.hexlify(byte_msg)
+            msg = hex_msg.decode("utf-8")
+
+            # Address
+            from_to_address = wallet.get_address()
+
+            # Store tx data to compare with queried one later.
+            channel_name = f"channel_{channel_order}"
+            TestLoopchain.sent_tx_data[channel_name] = {
+                "from": from_to_address,
+                "to": from_to_address,
+                "msg": msg
+            }
+
+            # Build transaction and sign it with wallet
+            transaction_data = {
+                "from": from_to_address,
+                "to": from_to_address,
+                "step_limit": 100000000,
+                "nid": 3,
+                "nonce": 100,
+                "data": msg,
+            }
+            transaction = MessageTransactionBuilder().from_dict(transaction_data).build()
+            signed_transaction = SignedTransaction(transaction, wallet)
+
+            # Send tx
+            url = f"http://localhost:9000/api/v3/{channel_name}"
+            print("Req url: ", url)
+            icon_service = IconService(HTTPProvider(url))
+            tx_hash = icon_service.send_transaction(signed_transaction)
+            print("Tx hash: ", tx_hash)
+
+            assert tx_hash.startswith("0x")
+            TestLoopchain.tx_hash_by_channel[channel_name] = tx_hash
+
+            await_sec = 0.5 * len(peer_conf_path_list)
+            print(f"Await consensus...({await_sec})")
+            time.sleep(await_sec)
```

Comment:
hmm.. I encountered a problem that a consensus is delayed when I tried to test in case of numerous channels and a quite large number of peers.

I understand that running multiple peers in poor machine could end to undesirable result. So  I decided to wait for a moment in every request to ensure consensus.